src/cuga/backend/server/manage_routes.py

```
@@ -881,11 +961,136 @@ async def save_manage_config_publish(request: Request, agent_id: Optional[str] =
 
-        ver = await save_config(config, agent_id)
+        # Compute vector-config hash and embed it in the config before saving.
+        # This hash covers only fields that affect stored vectors (embedding
+        # model, chunk size, metric) so versions with identical vector settings
+        # share the same collection.
+        from cuga.backend.knowledge.config import KnowledgeConfig as _KC
+
+        try:
+            _kb_obj = _KC.coerce_and_validate(knowledge_cfg)
+            _vec_hash = _kb_obj.vector_config_hash()
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)